packages/preview-middleware-client/src/cpe/changes/service.ts

```diff
@@ -303,25 +328,39 @@ export class ChangeService {
             this.changesRequiringReload = changesRequiringReload;
 
             if (Array.isArray(allCommands) && allCommands.length === 0) {
+                this.pendingChanges = [];
                 await this.fetchSavedChanges();
             }
-
-            this.updateStack(activeChanges);
+            this.updateStack();
             handleStackChange();
         };
     }
 
-    private prepareChangeType(
+    private async handleCommand(command: FlexCommand, inactiveCommandCount: number, index: number): Promise<void> {
+        const pendingChange = await this.prepareChangeType(command, inactiveCommandCount, index);
+        if (pendingChange) {
+            this.pendingChanges.push(pendingChange);
```

Comment:
There seems to be an issue with pending changes duplicating on each event. It is probably because `pendingChanges` field is not reset and we always process all the changes in the stack.